server.js

```diff
@@ -15,16 +15,217 @@ import cors from "cors";
+
+    if (top20Popular.length > 0) {
+        res.json(top20Popular);
+    } else {
+        res.status(404).send("No popular books found");
+    }
+});
+
+app.get("/books/highest-rated", (req, res) => {
+    const highestRatedBooks = [...booksData].sort(
+        (a, b) => b.average_rating - a.average_rating
+    );
+
+    const top20Rated = highestRatedBooks.slice(0, 20);
+
+    if (top20Rated.length > 0) {
+        res.json(top20Rated);
+    } else {
+        res.status(404).send("No highest-rated books found");
+    }
+});
+
+// creating category short stories, using Math random to select 20 items from the array of books.
+app.get("/books/short-stories", (req, res) => {
+    const shortStories = booksData.filter((book) => book.num_pages < 100);
+
+    // Shuffle the array randomly using Fisher-Yates algorithm
+    /* Fisher-Yates: randomizing the sort of the books array futher.
+        Compares pairs of books. Instead of comparing them in a regular order, 
+        we subtract a random value between -0.5 and 0.5 from each comparison.
+        If the result is negative, one book goes before the other.
+        If the result is positive, the other book goes first.
+        If the result is zero, their order doesn't change.*/
+    const shuffledShortStories = shortStories.sort(() => Math.random() - 0.5);
+
+    // Return the first 20 items
+    const randomShortStories = shuffledShortStories.slice(0, 20);
+
+    if (randomShortStories.length > 0) {
+        res.json(randomShortStories);
+    } else {
+        res.status(404).send("No novels found");
+    }
+});
+
+app.get("/books/novels", (req, res) => {
+    const novels = booksData.filter((book) => book.num_pages > 100);
+
+    // Shuffle the array randomly using Fisher-Yates algorithm
+    const shuffledNovels = novels.sort(() => Math.random() - 0.5);
+
+    // Return the first 20 items
+    const randomNovels = shuffledNovels.slice(0, 20);
+
+    if (randomNovels.length > 0) {
+        res.json(randomNovels);
+    } else {
+        res.status(404).send("No novels found");
+    }
+});
+
+app.get("/books/authors/:authors", (req, res) => {
+    //defining and formatting the slur (authors name)
+    const targetAuthor = req.params.authors.toLowerCase().replace(/\s/g, "");
+
+    let booksByAuthors = booksData.filter((item) => {
+        //formatting author names to be able to put them in url
+        const formattedItemAuthors = item.authors
+            .toLocaleLowerCase()
+            .replace(/\s/g, "");
+
+        //returning all books where targetAuthor is included as author
+        return formattedItemAuthors.includes(targetAuthor);
+    });
+
+    //checking that user found an array with the targetAuthor they defined
+    if (booksByAuthors.length > 0) {
+        res.json(booksByAuthors);
+    } else {
+        //if no match is found, returning an error message.
+        res.status(404).send("No book found by that author");
+    }
+});
+
+//DUMMY/EMPTY ENDPOINTS
```

Comment:
I like your ambitious dummy endpoints and that you give a lot of context to how they would be written in code and why!